google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -294,25 +292,25 @@ private static ManagedChannel createChannel(String rootUrl, RpcChannelFactory fa
 
     @Override
     public ManagedChannel newChannel(String host, int port) {
-      NettyChannelBuilder builder =
-          NettyChannelBuilder.forAddress(host, port)
-            .sslContext(newSslContext())
+      ManagedChannelBuilder builder =
+          ManagedChannelBuilder.forAddress(host, port)
+            // .sslContext(newSslContext())
             .intercept(interceptors)
-            .maxHeaderListSize(MAX_HEADER_LIST_SIZE)
-            .maxMessageSize(MAX_MESSAGE_SIZE);
+            // .maxHeaderListSize(MAX_HEADER_LIST_SIZE)
+            .maxInboundMessageSize(MAX_MESSAGE_SIZE);
```

Comment:
@vkedia How bad is this?

For context, many people are having problems with netty version conflicts. So gRPC provided `grpc-netty-shaded` in gRPC 1.9. Using it helps with the conflict, but it means we can't/shouldn't use Netty here.

Reply:
Potentially we should leave Spanner with grpc-netty and file an issue to get it migrated to grpc-netty-shaded, so that we don't block everything else.

Reply:
Nevermind, I don't think this will work. Anyone who wanted to use both Pub/Sub and Spanner could only have one of grpc-netty or grpc-netty-shaded. We need to solve Spanner's problem here before we can switch over to grpc-netty-shaded.

Reply:
This cannot be removed since in certain cases cloud spanner would send a bigger header which will cause gRPC to throw an exception if if we do not increase the header size here. I think gprc should expose this functionality on ManagedChannelBuilder as well.

Reply:
@garrettjonesgoogle According to [gRPC release doc](https://github.com/grpc/grpc-java/releases/tag/v1.9.0), it is safe to have both `grpc-netty` and `grpc-netty-shaded` right?

Reply:
If someone needs to use both Pub/Sub and Spanner, and they also can't get `grpc-netty` working such that they need to use `grpc-netty-shaded`, then Spanner won't work for them even if it's ok to have both `grpc-netty` and `grpc-netty-shaded` on the classpath. @ejona86 , do I have it right?

Reply:
But anyway, it appears that we could improve the situation for non-Spanner while not making Spanner worse off.

Reply:
Reverted changes to spanner

Reply:
Yes it is safe to depend on both at the same time. I will mention that we don't specify a relative priority between the two; gRPC will just pick one and hope for the best.

Reply:
@ejona86 are there any plans to add these methods to ManagedChannelBuilder so that spanner can also move off using NettyChannelBuilder. Is there any other way we can do this without having to depend on NettyChannelBuilder. I would love to have spanner also depend on netty shaded since we have heard from multiple customers with this being an issue.

Reply:
@vkedia, by "these methods" do you mean only `maxHeaderListSize`? There's not currently any feature request open to add something like `maxHeaderListSize` to `ManagedChannelBuilder`. Maybe you want to create one?

I'll note that maxInboundMessageSize is currently experimental. https://github.com/grpc/grpc-java/issues/2563 is its tracking issue.

Reply:
Filed an issue: https://github.com/grpc/grpc-java/issues/4050